rest/config.go

```diff
@@ -837,3 +851,37 @@ func ServerMain(runMode SyncGatewayRunMode) {
 	ValidateConfigOrPanic(runMode)
 	RunServer(config)
 }
+
+// localhostAddr takes an address like ':4984' or '127.0.0.1:4984' and prefixes with localhost if required.
+func localhostAddr(addr string) string {
```

Comment:
What about writing a unit test with a variety of inputs and expected outputs?

I'd be curious to see what happens if it was configured with `localhost:4984`.  (not actually sure it's a valid input for config, but I would expect this code to at least avoid trying to do a GET request on localhostlocalhost:4984)

Reply:
Added unit test in `config_test.go`